nixos/modules/services/networking/ddclient.nix

```diff
@@ -31,8 +31,8 @@ let
   preStart = ''
     install ${configFile} /run/${RuntimeDirectory}/ddclient.conf
     ${lib.optionalString (cfg.configFile == null) (if (cfg.passwordFile != null) then ''
-      password=$(head -n 1 ${cfg.passwordFile})
-      sed -i "s/^password=$/password=$password/" /run/${RuntimeDirectory}/ddclient.conf
+      password=$(printf "%q" $(head -n 1 ${cfg.passwordFile}))
```

Comment:
Suggestion: add quotes to not break password with space:

`password=$(printf "%q" "$(head -n 1 "${cfg.passwordFile}")")`

(And quotes around the passwordFile in case someone is crazy enough to use spaces in the filename :-D)

Reply:
Done.